### ISSUE TITLE
[2.3.0-RC1][Cypher] Unable to load node with ID 1 When deleting node

### ISSUE THREAD

**Issue by ikwattro** [neo4j/neo4j#5653]
Hi,

Fresh Install of RC1.

Execute Cypher : 

```
CREATE (:Person {name:'Michal'})-[:FRIEND_OF {since:2007}]->(:Person {name:'Daniela'})
```

Then try to MATCH the whole pattern and delete it : 

```
MATCH (p1:Person {name:'Michal'})-[r:FRIEND_OF {since:2007}]->(p2:Person {name:'Daniela'}) DELETE r, p1, p2
```

Result in the following error : 

```
Unable to load NODE with id 1.
```

Result is same in the browser and in our testing suite.

Seems related to issue #5598 

Stacktrace : 

```
org.neo4j.kernel.api.exceptions.EntityNotFoundException: Unable to load NODE with id 1.
    at org.neo4j.kernel.impl.api.StateHandlingStatementOperations.nodeCursorById(StateHandlingStatementOperations.java:138)
    at org.neo4j.kernel.impl.api.ConstraintEnforcingEntityOperations.nodeCursorById(ConstraintEnforcingEntityOperations.java:394)
    at org.neo4j.kernel.impl.api.OperationsFacade.nodeGetRelationships(OperationsFacade.java:317)
    at org.neo4j.cypher.internal.spi.v2_3.TransactionBoundQueryContext.getRelationshipsForIds(TransactionBoundQueryContext.scala:146)
    at org.neo4j.cypher.internal.compiler.v2_3.spi.DelegatingQueryContext.getRelationshipsForIds(DelegatingQueryContext.scala:63)
    at org.neo4j.cypher.internal.compiler.v2_3.spi.DelegatingQueryContext.getRelationshipsForIds(DelegatingQueryContext.scala:63)
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.matching.SingleStep$$anonfun$1.apply(SingleStep.scala:46)
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.matching.SingleStep$$anonfun$1.apply(SingleStep.scala:45)
    at org.neo4j.cypher.internal.compiler.v2_3.helpers.DynamicIterable$$anon$1.iterator(DynamicIterable.scala:29)
    at org.neo4j.cypher.internal.compiler.v2_3.helpers.DynamicJavaIterable$$anon$3.iterator(DynamicIterable.scala:39)
    at org.neo4j.kernel.impl.traversal.TraversalBranchWithState.expandRelationshipsWithoutChecks(TraversalBranchWithState.java:71)
    at org.neo4j.kernel.impl.traversal.TraversalBranchImpl.expandRelationships(TraversalBranchImpl.java:104)
    at org.neo4j.kernel.impl.traversal.StartNodeTraversalBranch.next(StartNodeTraversalBranch.java:47)
    at org.neo4j.kernel.impl.traversal.AsOneStartBranch.next(AsOneStartBranch.java:109)
    at org.neo4j.graphdb.traversal.PreorderDepthFirstSelector.next(PreorderDepthFirstSelector.java:49)
    at org.neo4j.kernel.impl.traversal.MonoDirectionalTraverserIterator.fetchNextOrNull(MonoDirectionalTraverserIterator.java:65)
    at org.neo4j.kernel.impl.traversal.MonoDirectionalTraverserIterator.fetchNextOrNull(MonoDirectionalTraverserIterator.java:36)
    at org.neo4j.helpers.collection.PrefetchingIterator.peek(PrefetchingIterator.java:60)
    at org.neo4j.helpers.collection.PrefetchingIterator.hasNext(PrefetchingIterator.java:46)
    at scala.collection.convert.Wrappers$JIteratorWrapper.hasNext(Wrappers.scala:42)
    at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:396)
    at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:396)
    at scala.collection.Iterator$$anon$13.hasNext(Iterator.scala:413)
    at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:396)
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.EmptyResultPipe.internalCreateResults(EmptyResultPipe.scala:30)
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.PipeWithSource.createResults(Pipe.scala:127)
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.DefaultExecutionResultBuilderFactory$ExecutionWorkflowBuilder.createResults(DefaultExecutionResultBuilderFactory.scala:93)
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.DefaultExecutionResultBuilderFactory$ExecutionWorkflowBuilder.build(DefaultExecutionResultBuilderFactory.scala:63)
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.ExecutionPlanBuilder$$anonfun$getExecutionPlanFunction$1.apply(ExecutionPlanBuilder.scala:230)
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.ExecutionPlanBuilder$$anonfun$getExecutionPlanFunction$1.apply(ExecutionPlanBuilder.scala:214)
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.ExecutionPlanBuilder$$anon$3.run(ExecutionPlanBuilder.scala:172)
    at org.neo4j.cypher.internal.compatibility.CompatibilityFor2_3$ExecutionPlanWrapper$$anonfun$run$1.apply(CompatibilityFor2_3.scala:192)
    at org.neo4j.cypher.internal.compatibility.CompatibilityFor2_3$ExecutionPlanWrapper$$anonfun$run$1.apply(CompatibilityFor2_3.scala:192)
    at org.neo4j.cypher.internal.compatibility.exceptionHandlerFor2_3$.runSafely(CompatibilityFor2_3.scala:114)
    at org.neo4j.cypher.internal.compatibility.CompatibilityFor2_3$ExecutionPlanWrapper.run(CompatibilityFor2_3.scala:191)
    at org.neo4j.cypher.internal.PreparedPlanExecution.execute(PreparedPlanExecution.scala:27)
    at org.neo4j.cypher.ExecutionEngine.execute(ExecutionEngine.scala:118)
    at org.neo4j.cypher.ExecutionEngine.execute(ExecutionEngine.scala:112)
    at org.neo4j.cypher.javacompat.internal.ServerExecutionEngine.executeQuery(ServerExecutionEngine.java:62)
    at org.neo4j.kernel.impl.factory.GraphDatabaseFacade.execute(GraphDatabaseFacade.java:333)
    at org.neo4j.kernel.impl.factory.GraphDatabaseFacade.execute(GraphDatabaseFacade.java:323)
```

**Comment by joshdmiller** [comment]
I am encountering a similar error with a query that works in 2.2 but gives this error in 2.3. Is there any update or progress on this issue?

**Comment by ikwattro** [comment]
I think there are a lot of similar issues, which might be fixed already and prepared for the 2.3.1 release.

**Comment by rickystillwell** [comment]
I am also hitting this with the neo4j-elasticsearch integration when trying to use with Neo4j 2.3.1

`2015-11-30 13:31:04.350-0600 ERROR Failed to commit transaction. org.neo4j.kernel.api.exceptions.TransactionFailureException: 
java.lang.RuntimeException: org.neo4j.kernel.api.exceptions.TransactionFailureException: 
    at org.neo4j.server.rest.transactional.TransitionalTxManagementKernelTransaction.commit(TransitionalTxManagementKernelTransaction.java:87)
    at org.neo4j.server.rest.transactional.TransactionHandle.closeContextAndCollectErrors(TransactionHandle.java:279)
    at org.neo4j.server.rest.transactional.TransactionHandle.commit(TransactionHandle.java:148)
    at org.neo4j.server.rest.web.TransactionalService$2.write(TransactionalService.java:211)
    at com.sun.jersey.core.impl.provider.entity.StreamingOutputProvider.writeTo(StreamingOutputProvider.java:71)
    at com.sun.jersey.core.impl.provider.entity.StreamingOutputProvider.writeTo(StreamingOutputProvider.java:57)
    at com.sun.jersey.spi.container.ContainerResponse.write(ContainerResponse.java:302)
    at com.sun.jersey.server.impl.application.WebApplicationImpl._handleRequest(WebApplicationImpl.java:1510)
    at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1419)
    at com.sun.jersey.server.impl.application.WebApplicationImpl.handleRequest(WebApplicationImpl.java:1409)
    at com.sun.jersey.spi.container.servlet.WebComponent.service(WebComponent.java:409)
    at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:558)
    at com.sun.jersey.spi.container.servlet.ServletContainer.service(ServletContainer.java:733)
    at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
    at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:800)
    at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1669)
    at org.neo4j.server.rest.web.CollectUserAgentFilter.doFilter(CollectUserAgentFilter.java:69)
    at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
    at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585)
    at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:221)
    at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1125)
    at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
    at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
    at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1059)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
    at org.eclipse.jetty.server.handler.HandlerList.handle(HandlerList.java:52)
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
    at org.eclipse.jetty.server.Server.handle(Server.java:497)
    at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:310)
    at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:248)
    at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540)
    at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:620)
    at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:540)
    at java.lang.Thread.run(Thread.java:745)
Caused by: org.neo4j.kernel.api.exceptions.TransactionFailureException: 
    at org.neo4j.kernel.impl.api.KernelTransactionImplementation.commit(KernelTransactionImplementation.java:514)
    at org.neo4j.kernel.impl.api.KernelTransactionImplementation.close(KernelTransactionImplementation.java:452)
    at org.neo4j.server.rest.transactional.TransitionalTxManagementKernelTransaction.commit(TransitionalTxManagementKernelTransaction.java:83)
    ... 33 more
Caused by: org.neo4j.kernel.api.exceptions.TransactionHookException: Transaction handler failed.
    at org.neo4j.kernel.impl.api.TransactionHooks$TransactionHooksState.add(TransactionHooks.java:100)
    at org.neo4j.kernel.impl.api.TransactionHooks.beforeCommit(TransactionHooks.java:59)
    at org.neo4j.kernel.impl.api.KernelTransactionImplementation.commit(KernelTransactionImplementation.java:512)
    ... 35 more
Caused by: org.neo4j.graphdb.NotFoundException: Node not found
    at org.neo4j.kernel.impl.core.NodeProxy.getLabels(NodeProxy.java:698)
    at org.neo4j.elasticsearch.ElasticSearchEventHandler.hasLabel(ElasticSearchEventHandler.java:94)
    at org.neo4j.elasticsearch.ElasticSearchEventHandler.beforeCommit(ElasticSearchEventHandler.java:53)
    at org.neo4j.elasticsearch.ElasticSearchEventHandler.beforeCommit(ElasticSearchEventHandler.java:29)
    at org.neo4j.kernel.TransactionEventHandlers.beforeCommit(TransactionEventHandlers.java:130)
    at org.neo4j.kernel.TransactionEventHandlers.beforeCommit(TransactionEventHandlers.java:49)
    ... 37 more
Caused by: org.neo4j.kernel.api.exceptions.EntityNotFoundException: Unable to load NODE with id 2.
    at org.neo4j.kernel.impl.api.StateHandlingStatementOperations.nodeCursorById(StateHandlingStatementOperations.java:138)
    at org.neo4j.kernel.impl.api.ConstraintEnforcingEntityOperations.nodeCursorById(ConstraintEnforcingEntityOperations.java:394)
    at org.neo4j.kernel.impl.api.OperationsFacade.nodeGetLabels(OperationsFacade.java:278)
    at org.neo4j.kernel.impl.core.NodeProxy.getLabels(NodeProxy.java:687)
    ... 42 more
`

**Comment by ikwattro** [comment]
Which ES integration are you using?

Envoyé de mon iPhone

**Comment by rickystillwell** [comment]
https://github.com/neo4j-contrib/neo4j-elasticsearch

**Comment by jotomo** [comment]
@ikwattro Doesn't occur anymore on my machine when using 2.3.1.

**Comment by craigtaverner** [comment]
Fixed in 2.3.0
